### PR TITLE
HTML tag fallbacks and remove Result around parser

### PR DESCRIPTION
**Pull request by dillonkearns**
This PR makes `Markdown.Parser.parse` infallible — it returns `List Block` instead of `Result (List DeadEnd) (List Block)`. Malformed input is treated as plain text (paragraphs), matching how mature markdown parsers handle arbitrary input.

## New

[`tryRender`](https://elm-doc-preview.netlify.app/Markdown-Renderer?repo=dillonkearns%2Felm-markdown&version=html-fallback#tryRender) works with any `Renderer err view` and returns `Result err (List view)`. The `render`/`tryRender` split mirrors `Task.perform`/`Task.attempt` — `render` is for the case where failure is impossible, `tryRender` is for when it's not.

```diff
+ tryRender : Renderer err view -> List Block -> Result err (List view)
```

[`Markdown.Html.withFallback`](https://elm-doc-preview.netlify.app/Markdown-Html?repo=dillonkearns%2Felm-markdown&version=html-fallback#withFallback) converts a fallible HTML renderer into an infallible one by providing a fallback for unmatched tags. This is the bridge from `Renderer String` to `Renderer Never`.

```diff
+ withFallback :
+     (String -> List { name : String, value : String } -> List view -> view)
+     -> Renderer String (List view -> view)
+     -> Renderer Never (List view -> view)
```

## Changed

[`Markdown.Parser.parse`](https://elm-doc-preview.netlify.app/Markdown-Parser?repo=dillonkearns%2Felm-markdown&version=html-fallback#parse) returns `List Block` directly. `deadEndToString` is removed.

```diff
- parse : String -> Result (List (Advanced.DeadEnd String Parser.Problem)) (List Block)
+ parse : String -> List Block
```

[`Renderer`](https://elm-doc-preview.netlify.app/Markdown-Renderer?repo=dillonkearns%2Felm-markdown&version=html-fallback#Renderer) gains an `err` type parameter to track whether the HTML renderer can fail.

```diff
- type alias Renderer view = { ... }
+ type alias Renderer err view = { ... }
```

[`render`](https://elm-doc-preview.netlify.app/Markdown-Renderer?repo=dillonkearns%2Felm-markdown&version=html-fallback#render) now requires `Renderer Never view` and returns `List view` (no `Result`).

```diff
- render : Renderer view -> List Block -> Result String (List view)
+ render : Renderer Never view -> List Block -> List view
```

## Spec compliance

5 additional CommonMark/GFM spec tests now pass (malformed Raw HTML examples [618](https://spec.commonmark.org/0.30/#example-618), [619](https://spec.commonmark.org/0.30/#example-619), [620](https://spec.commonmark.org/0.30/#example-620), [621](https://spec.commonmark.org/0.30/#example-621), [632](https://spec.commonmark.org/0.30/#example-632)).

## Test coverage

- Fuzz tests targeting markdown hot spots (character-level, fragment-level, document-level) verify the parser never crashes
- Full parse+render pipeline fuzz tests with `withFallback`
- Edge case unit tests: unterminated code fences, CRLF, BOM, control characters, deeply nested block quotes

### CODE REVIEW

File: examples/src/Slugs.elm [hunk 1]
```diff
@@ -60,7 +56,7 @@ markdownInputView markdownInput =
 
 specials : Regex.Regex
 specials =
-    "[\u{2000}-\u{206F}⸀-\u{2E7F}\\\\'!\"#$%&()*+,./:;<=>?@[\\\\]^`{|}~’]"
+    "[\u{2000}-\u{206F}⸀-\u{2E7F}\\\\'!\"#$%&()*+,./:;<=>?@[\\\\]^`{|}~']"
```
Comment (miniBill) [inline examples/src/Slugs.elm]: Are you sure you want this one? `'` is already in the regex, afaict

File: spec-tests/OutputMarkdownHtml.elm [hunk 2]
```diff
@@ -42,14 +41,7 @@ init flags =
...
         |> Markdown.parse
-        |> Result.mapError deadEndsToString
-        |> Result.andThen (\ast -> Markdown.Renderer.render renderer ast)
-
-
-deadEndsToString deadEnds =
-    deadEnds
-        |> List.map Markdown.deadEndToString
-        |> String.join "\n"
+        |> (\ast -> Markdown.Renderer.render renderer ast)
```
Comment (miniBill) [inline spec-tests/OutputMarkdownHtml.elm]: 💅 this can be simplified

File: examples/src/Slugs.elm (hunk 1 above)
Comment (dillonkearns) [reply examples/src/Slugs.elm]: Thank you!

File: spec-tests/OutputMarkdownHtml.elm (hunk 2 above)
Comment (dillonkearns) [reply spec-tests/OutputMarkdownHtml.elm]: Thank you!
